### PR TITLE
OutboxHas

### PR DESCRIPTION
**Pull request by brian-js**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/safe_vault/321)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @Fraser999

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/personas/mpid_manager.rs [hunk 1]
```diff
@@ -315,50 +318,77 @@ impl MpidManager {
...
+                if let Some(ref account) = self.accounts.get(&request.dst.get_name().clone()) {
+                    let names_in_outbox = header_names.iter()
+                                                      .filter(|h| account.has_in_outbox(h))
+                                                      .cloned()
+                                                      .collect::<Vec<XorName>>();
+                    let mut mpid_headers = vec![];
+
+                    for name in names_in_outbox.iter() {
+                        if let Ok(data) = self.chunk_store_outbox.get(name) {
+                            let mpid_message: MpidMessage = unwrap_result!(deserialise(&data));
```
Comment (maqi) [inline src/personas/mpid_manager.rs]: the for loop will be broken as long as there is any one corrupted data.
probably we have to use a match to ensure the for loop can tolerate corrupted data.

Comment (brian-js) [inline src/personas/mpid_manager.rs]: If it's anything other than Ok it'll loop to the next name, IIUC.

File: src/personas/mpid_manager.rs [hunk 2]
```diff
@@ -315,50 +318,77 @@ impl MpidManager {
...
                                     client.clone(), Data::PlainData(data.clone()), message_id.clone());
                             }
                         }
                     }
-                    None => warn!("can not find the account {:?}", dst.get_name()),
+                    None => warn!("can not find the account {:?}", request.dst.get_name().clone()),
+                }
+            }
+            MpidMessageWrapper::OutboxHas(header_names) => {
+                if let Some(ref account) = self.accounts.get(&request.dst.get_name().clone()) {
```
Comment (maqi) [inline src/personas/mpid_manager.rs]: do we need to check the src requester is the correct owner of the account (src being one of the registered client)?
the current implementation allows anyone query other's outbox

File: src/personas/mpid_manager.rs [hunk 3]
```diff
@@ -83,7 +83,7 @@ impl MailBox {
     }
 
     #[allow(dead_code)]
```
Comment (maqi) [inline src/personas/mpid_manager.rs]: seems the allow dead_code can be removed

**Pull request merged**